src/Mafia/Cabal/Version.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Mafia.Cabal.Version
+  ( getCabalVersion
+  , checkCabalVersion
+  ) where
+
+import           Control.Monad.IO.Class (liftIO)
+
+import           Data.Text (Text)
+import qualified Data.Text as T
+
+import           Mafia.Cabal.Process
+import           Mafia.Cabal.Types
+import           Mafia.Package
+import           Mafia.Process
+
+import           P
+
+import           System.IO (IO)
+
+import           X.Control.Monad.Trans.Either (EitherT, left, runEitherT)
+
+
+getCabalVersion :: EitherT CabalError IO Version
+getCabalVersion = do
+  result <- liftIO . runEitherT $ cabal "--version" []
+  case result of
+    Left  _         -> left CabalNotInstalled
+    Right (Out out) ->
+      case parseCabalVersion out of
+        Nothing  -> left (CabalCouldNotParseVersion out)
+        Just ver -> return ver
+
+parseCabalVersion :: Text -> Maybe Version
+parseCabalVersion out =
+  case T.words out of
+    ("cabal-install" : "version" : version : _) -> parseVersion version
+    _                                           -> Nothing
+
+-- TODO not sure if maybe we should have no upper bounds here?
+-- TODO maybe mafia should just install its own cabal?
```

Comment:
I think installing own cabal is tremendously dangerous/not a good idea. As it stands using cabal and mafia at the same time is standard faire, and risking inconsistent versions could have particularly dire consequences. At least it the naive version of just installing its own and using it regardless of what is on path.